src/components/SideNavigation/SideNavigation.tsx

```diff
@@ -0,0 +1,140 @@
+
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import Icon from '../Icon';
+import { SIDENAVIGATION } from '../ThemeIdentifiers';
+import { Drawer, DrawerContent } from '../Drawer';
+import Button from '../Button';
+import Tooltip from '../Tooltip';
+import Accordion from '../Accordion';
+import {AccordionItemProps} from '../Accordion';
+import * as baseTheme from './SideNavigation.scss';
+
+export interface INavigationData {
+    id?:number;
+    label?: React.ReactNode;
+    icon?: React.ReactNode;
+    children?: React.ReactNode;
+    action?(): void;
+}
+
+export interface Props {
+    theme?: any;
+    accordian?: boolean,
+    source: INavigationData[],
+}
+
+export interface State {
+    drawer: boolean;
+    activeDrawerId: string;
+}
+
+class SideNavigation extends React.Component<Props, State> {
+    constructor(props: Props) {
+      super(props);
+      this.state = {
+        drawer: false,
+        activeDrawerId: 'fullContent',
+      };
+    }
+
+    toggleDrawer = () => {
+        this.setState({ drawer: !this.state.drawer });
+    }
+    componentDidUpdate(){
+        const bodyElement = document.body;
+        if (bodyElement !== null) {
+            bodyElement.className +=  this.state.drawer ? ' ' + this.props.theme.container : '';
+        }
+    }
+
+    render() {
+        const { source, theme, accordian } = this.props;
+        let actDrawerId = this.state.activeDrawerId;
+        const rootElement = document.getElementById('root');
+        const iconClass = theme.icon;
```

Comment:
@ishani-m Please destructure theme values as well instead of creating every single variables, it will be something like: `const { collapseIcon, navDivider, li: liClass } = theme;`, so if you want to rename the variable you can write like `li: liClass`